### PR TITLE
Fix (maybe) MongoDB tests flakiness

### LINKED ISSUE #13619: `TestAccessMongoDB` flakiness
## Failure
Saw this fail off of `branch/v10`.

### CI Logs
- https://console.cloud.google.com/cloud-build/builds/64878a7f-6064-4135-aeb7-24f155b9e04b;step=0?project=ci-account
- https://console.cloud.google.com/cloud-build/builds/61625553-453f-46e0-ad31-7a0e5495201a?project=ci-account
- https://console.cloud.google.com/cloud-build/builds/7fe827fa-b190-4db3-988e-454ed92bb2e0?project=ci-account
- https://console.cloud.google.com/cloud-build/builds;region=us-west1/b8812649-6701-4019-a687-b37cad789d22?project=ci-account
- https://console.cloud.google.com/cloud-build/builds;region=us-west1/1d1d3095-984a-4a8e-9091-bd97a8f6be9c?project=ci-account
- https://console.cloud.google.com/cloud-build/builds;region=us-west1/1d1d3095-984a-4a8e-9091-bd97a8f6be9c?project=ci-account

### Relevant Snippet
```
    --- FAIL: TestAccessMongoDB/access_denied_to_specific_user_and_database (13.73s)
===================================================
OUTPUT github.com/gravitational/teleport/lib/srv/db.TestAccessMongoDB/access_denied_to_specific_user_and_database/old_server/client_without_compression
===================================================
=== RUN   TestAccessMongoDB/access_denied_to_specific_user_and_database/old_server/client_without_compression
2022-06-17T15:20:41Z DEBU [DB:SERVIC] Asked check out of cycle srv/heartbeat.go:287
2022-06-17T15:20:41Z DEBU [DB:PROXY]  Available databases in root.example.com: [DatabaseServer(Name=mongo, Version=10.0.0-alpha.1, Hostname=teleport.cluster.local, HostID=42b1c357-3c83-409e-a28c-6114b95346e3, Database=Database(Name=mongo, Type=self-hosted, Labels=map[echo:test]))]. db/proxyserver.go:629
2022-06-17T15:20:41Z DEBU [AUTH]      ClientCertPool -> cert(root.example.com issued by root.example.com:4408681756169843927285952159426259152) auth/middleware.go:679
=== PAUSE TestAccessMongoDB/access_denied_to_specific_user_and_database/old_server/client_without_compression
=== CONT  TestAccessMongoDB/access_denied_to_specific_user_and_database/old_server/client_without_compression
=== CONT  TestAccessMongoDB/access_denied_to_specific_user_and_database/old_server/client_without_compression
    access_test.go:829: 
        	Error Trace:	access_test.go:829
        	Error:      	Received unexpected error:
        	            	server selection error: server selection timeout, current topology: { Type: Single, Servers: [{ Addr: 127.0.0.1:46073, Type: Unknown, Average RTT: 0 }, ] }
        	Test:       	TestAccessMongoDB/access_denied_to_specific_user_and_database/old_server/client_without_compression
```

### PR DESCRIPTION
**Pull request by r0mant**
During troubleshooting, I came to the conclusion that the main issue is that MongoDB tests has a ton of combinations (test cases + server options + client options), each creating a new environment from scratch and running in parallel, which probably starves some resource resulting in timeouts. The tests were failing even on my M1 Max Macbook.

So this PR makes a couple of tweaks:

- Set up test environment once and use it for all test cases.
- Bump some channel buffer sizes as I noticed e.g. some tests were blocking on a small "events" channel which also could've contributed to timeouts.

With these changes tests consistently pass for me on my Macbook and even on a t2.medium instance:

```
ubuntu@ip-172-31-20-136:~/teleport$ while go test -timeout 600s -count 1 -run ^TestAccessMongoDB github.com/gravitational/teleport/lib/srv/db; do :; done
ok  	github.com/gravitational/teleport/lib/srv/db	46.709s
ok  	github.com/gravitational/teleport/lib/srv/db	46.819s
ok  	github.com/gravitational/teleport/lib/srv/db	48.195s
ok  	github.com/gravitational/teleport/lib/srv/db	48.624s
ok  	github.com/gravitational/teleport/lib/srv/db	47.268s
ok  	github.com/gravitational/teleport/lib/srv/db	45.211s
ok  	github.com/gravitational/teleport/lib/srv/db	47.127s
ok  	github.com/gravitational/teleport/lib/srv/db	46.694s
ok  	github.com/gravitational/teleport/lib/srv/db	46.107s
ok  	github.com/gravitational/teleport/lib/srv/db	45.578s
ok  	github.com/gravitational/teleport/lib/srv/db	47.256s
ok  	github.com/gravitational/teleport/lib/srv/db	48.077s
ok  	github.com/gravitational/teleport/lib/srv/db	47.699s
ok  	github.com/gravitational/teleport/lib/srv/db	46.473s
ok  	github.com/gravitational/teleport/lib/srv/db	46.082s
ok  	github.com/gravitational/teleport/lib/srv/db	48.043s
ok  	github.com/gravitational/teleport/lib/srv/db	47.989s
```

Fixes https://github.com/gravitational/teleport/issues/13619.

### CODE REVIEW

**Comment by smallinsky** [comment]
Seems that flakiness is still present - current [build](https://github.com/gravitational/teleport/pull/14502/checks?check_run_id=7351245382) for this PR failed at:  
```
=== CONT  TestAccessMongoDB/no_access_to_databases/mongo-old/client_without_compression
    access_test.go:807: 
        	Error Trace:	access_test.go:807
        	Error:      	"server selection error: server selection timeout, current topology: { Type: Single, Servers: [{ Addr: 127.0.0.1:43335, Type: Unknown, Average RTT: 0 }, ] }" does not contain "access to db denied"
        	Test:       	TestAccessMongoDB/no_access_to_databases/mongo-old/client_without_compression
  ```
  
I'm debugging the Mongo driver internals  found the the issue is when the db mongo engine dials the Mongo TestServer in order to obtain Topology: 
```
[4d7bbb4] MYDEBUG Execute roundTrip conn add: 127.0.0.1:7771
[4d7bbb4] ----> roundTrip write message
[4d7bbb4] ----> roundTrip read message 228.783µs
[4d7bbb4] ----> roundTrip after message after 2.31437706s
[4d7bbb4] ----> roundTrip inside 2.314437021s
[4d7bbb4] MYDEBUG roundTrip end time 2.314641204s
```
This call sometimes takes > 5 seconds and depends on test execution env I haven't found the root cause yet but the timeout currently is  hardcoded in our code and set to[ 5 second ](https://github.com/gravitational/teleport/blob/master/lib/srv/db/common/constants.go#L24) For now I think that we should increate this timeout in our unit test till we will find the root cause.

**Comment by r0mant** [comment]
@smallinsky Isn't it ironic that this PR that was supposed to fix MongoDB tests flakiness failed due to MongoDB tests flakiness? :lolsob:

I would still like for these changes to land because they make tests more efficient (no spinning up extra env for each case). I'll troubleshoot a bit more and see if I can find if there's another bug. Maybe we can combine your fix and mine?

File: lib/reversetunnel/fake.go [hunk 1]
```diff
@@ -70,7 +70,7 @@ type FakeRemoteSite struct {
 func NewFakeRemoteSite(clusterName string, accessPoint auth.RemoteProxyAccessPoint) *FakeRemoteSite {
 	return &FakeRemoteSite{
 		Name:        clusterName,
-		connCh:      make(chan net.Conn),
+		connCh:      make(chan net.Conn, 100000),
```
Comment (nklaassen) [inline lib/reversetunnel/fake.go:73]: the other one got bumped up to ten thousand, did you mean to put one hundred thousand here? Seems high 🤷‍♂️

**Review by nklaassen — APPROVED** [review]
have seen a bunch of hits today, will probably be good to get this in

**Review by smallinsky — APPROVED** [review]

**Comment by r0mant** [comment]
Seems this actually makes things worse for whatever reason, 2nd attempt failed on Mongo tests too.

Let's first merge @smallinsky's fix that fixes the keypair generation and then we can try and take these in as well.

**Pull request closed without merging**